src/playlist.js

```diff
@@ -347,44 +348,58 @@ export const getMediaInfoForTime = function(
   startIndex,
   startTime
 ) {
-  let i;
-  let segment;
-  const numSegments = playlist.segments.length;
+
+  const partSegments = playlist.segments.reduce((acc, segment, si) => {
```

Comment:
We need this function to know about parts now. So we break put all of our segments and or parts into one list.